R/agglomerate.R

```diff
@@ -212,6 +218,30 @@
 #' merged <- agglomerateByVariable(
 #'     GlobalPatterns, by = "cols", colData(GlobalPatterns)$SampleType)
 #' merged
+#' 
+#' ## Agglomerate by multiple modules
+#' 
+#' # Generate 30 random modules
+#' N_module <- 30L
+#' modules <- sample(
+#'     c(TRUE, FALSE),
+#'     size = nrow(tse) * N_module,
+#'     prob = c(0.2, 0.8),
+#'     replace = TRUE
+#' )
+#'
+#' # Convert modules to matrix
+#' modules <- modules |>
+#'     matrix(nrow = nrow(tse))
+#' # Add feature names as rownames
+#' rownames(modules) <- rownames(tse)
+#' # Add module names as colnames
+#' colnames(modules) <- paste0("module_", seq_len(ncol(modules)))
+#' # Store modules table in metadata slot
+#' metadata(tse)$modules <- modules
+#' 
+#' # Agglomerate based on modules
+#' tse_module <- agglomerateByModule(tse, by = 1, group = "modules")
```

Comment:
should we also show in the example how one could add this to altExp directly (as an alternative to creating a completely new tse)?